CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -63,64 +60,56 @@ class CardClient internal constructor(
                 CheckoutOrdersAPI(configuration),
                 DataVaultPaymentMethodTokensAPI(configuration),
                 AnalyticsService(activity.applicationContext, configuration),
-                BrowserSwitchClient(),
+                CardAuthLauncher(),
                 Dispatchers.Main
             )
 
     init {
         activity.lifecycle.addObserver(lifeCycleObserver)
     }
 
+    // NEXT MAJOR VERSION: Consider renaming approveOrder() to confirmPaymentSource()
     /**
      * Confirm [Card] payment source for an order.
      *
      * @param activity [FragmentActivity] activity used to start 3DS flow (if requested)
      * @param cardRequest [CardRequest] for requesting an order approval
      */
     fun approveOrder(activity: FragmentActivity, cardRequest: CardRequest) {
+        // TODO: deprecate this method and offer auth challenge integration pattern (similar to vault)
         orderId = cardRequest.orderId
         analyticsService.sendAnalyticsEvent("card-payments:3ds:started", orderId)
 
         CoroutineScope(dispatcher).launch(approveOrderExceptionHandler) {
-            confirmPaymentSource(activity, cardRequest)
-        }
-    }
-
-    private suspend fun confirmPaymentSource(activity: FragmentActivity, cardRequest: CardRequest) {
```

Comment:
Had to inline this method to reduce the overall method count for this class to please the linter 🙌

Reply:
Would it be better to add an exception to the linter rule here if having this function separated increased readability? Or would splitting out some of this internal logic into a separate class make sense?

Reply:
I thought about this at the time. I'm honestly torn. The linter has been somewhat strict, but it does a good job of making us rethink the arch sometimes. This made me feel like maybe the `notify` methods at the bottom of this class could go somewhere else, like in a composable `CardAnalyticsClient` that uses an `AnalyticsService` reference internally.